hal/src/gpio/v2.rs

```diff
@@ -15,6 +15,8 @@
 //! between pins. However, by doing so, pins must now be tracked at run-time,
 //! and each pin has a non-zero memory footprint.
 
+#![allow(clippy::bool_comparison)]
```

Comment:
We can merge this, but I think I'm going to change this.

@vccggorski convinced me that `bool` is often used where there are better options, merely because there are only two choices. I think I will add a `LogicLevel` `enum` to represent `High` and `Low`. I think that will convert the `bool` comparisons and give clearer code.

Reply:
> @TDHolmes, I only looked through "my" modules. I didn't see any missing safety comments. What were you referring to?

@bradleyharden, specifically: `SpiFuture::free_unchecked`, `v2::spi::Config::sercom`, `v2::spi::Spi::read_data` and `v2::spi::Spi::write_data`.